### PR TITLE
[JSS] [React Native] Update packages, fix issues

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description
1. Updated packages.
2. `Date` component will always render `<HtmlView />` if `editable` is provided, previously it could use `render` prop even if `editable` is provided. [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-cf0eb2141a5287bad0bcffcc0c03093478a06c321d5619ec1a1cdc82ef48a9a7R32-L42)
3. Image component: apply `Object` style for svg image, instead of `Array`. SvgUri doesn't support array of styles. [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-be9ea919300f5a815788858dbb46d44433559bb1574aa2769329369184a253ebR83-R92)
4. Remove `dataConverstation` since `editable` property not used in `disconnected` mode. [[Fix]](https://github.com/Sitecore/jss/pull/624/files?file-filters%5B%5D=.config&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.svg&file-filters%5B%5D=.ts#diff-811b30e071c2324d1efc8b7df4177a747b69754a2189a9f9a12a0592e19db5eb)
5. StyleguideFieldUsageImage should use .png instead of .svg. [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-c019fbab5ea770fdb0debd59c31e9b691cd6a04ba5ab4877a84dc47defc5b1b0)
6. Clean android build before `start`, `react-native run-android` don't clean build, may be because of permissions on my machine. (build will fail if you don't clean it)  [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-74001dce5e90e71a361d716b5a0f511ffb37b28ae584bbeae822210bd4330561R20)
7. Added allowed media params which were missed. [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-589e2df10650baa8da2437df55e73e30c8c4232d8070f4950dec494319026b23)
8. Added shared `<Layout/>` component which should be rendered by every route, in order to have same navigation panel everywhere. Made Logo clickable, you can navigate to Home page [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-4af9628100b02fe7b72af24e7378f58fa24dc97fa7d62576a3f43c74ffb9d482)
9. StyleguideFieldUsageDate wasn't working in connected mode, because `render` prop should wrap string value by `<Text/>` [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-c26cc3e661120837e5c28886bd282db0e3f6c4b78f4cda2ec668a5f8ab63f27d)
10. Fixed incorrect `src` prop type for StyleguideFieldUsageImage. [[Fix]](https://github.com/Sitecore/jss/pull/624/files#diff-3df17d4bb29ef61323d3e083a28faf72dff628a57a874a38ecd0f93ac75fc6cd)
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [ ] I have read the Contributing guide.
- [ ] My code follows the code style of this project.
- [ ] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

**Review by ambrauer — APPROVED** [review]
Looks good (other than one comment on release note), but @illiakovalenko let's leave the branch unmerged until QA is done (since I'm doing that as well).

File: docs/data/routes/release-notes/en.md [hunk 1]
```diff
@@ -5,6 +5,25 @@ title: Release Notes
 ---
 # Release Notes
 
+## Sitecore JSS 16.1 for Sitecore 10
+
+### Breaking changes
+* [PR #624](https://github.com/Sitecore/jss/pull/624) [sitecore-jss-react-native] Removed `dataConversation`, since `editable` property not used in `disconnected` mode.
```
[the remark this answers is not in the document]
Comment (illiakovalenko) [reply docs/data/routes/release-notes/en.md:11]: @ambrauer It was used in the sample app, and probably somebody use this function and will not expect to see error.

File: samples/react-native/package.json [hunk 2]
```diff
@@ -17,25 +17,29 @@
...
     "start-ios:connected-tunnel": "scjss-verify-setup && node ./build/start.js --platform ios --appMode connected-tunnel",
     "lint": "eslint ./src/**/*.js ./sitecore/definitions/**/*.js ./**/*.js"
   },
   "dependencies": {
+    "@react-native-community/masked-view": "^0.1.10",
     "@sitecore-jss/sitecore-jss-react-native": "^16.1.0-canary.5",
     "axios": "^0.21.1",
     "prop-types": "^15.6.0",
     "react": "16.13.1",
     "react-native": "^0.63.4",
```
[the remark this answers is not in the document]
Comment (illiakovalenko) [reply samples/react-native/package.json:35]: Probably, let's leave for now - will upgrade in the next round of upgrading dependencies. In this case will need to spend additional time, in order to retest it at least

File: docs/data/routes/release-notes/en.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ambrauer) [reply docs/data/routes/release-notes/en.md:11]: @illiakovalenko Makes sense, let's leave it.

**Pull request merged**